libmscore/staff.cpp

```diff
@@ -1035,10 +1035,15 @@ void Staff::staffTypeListChanged(const Fraction& tick)
       else
             triggerLayout(Fraction::fromTicks(range.first));
 
-      if (range.second < 0)
-            triggerLayout(score()->lastMeasure()->endTick());
-      else
+      if (range.second < 0) {
```

Comment:
You be a one-liner `if (range.second < 0 && score()->lastMeasure()) ` maybe?

Reply:
No, because if one of the conditions is false, control will go to the <code>else</code> branch while it shouldn't. It turns this three state logic into a two state logic.

Reply:
OK, so there are cases where nothing needs to be done

Reply:
Yes, in this particular case ;-)
When reading a score **and** a staff change is in the first measure, this method is triggered but there is no measure, <code>lastMeasure()</code> has not valid <code>endTick()</code>, there is no layout to be triggered. In this case nothing has to be done.